infor-design/enterprise
# 1757 - Calendar Week and Day view

**Explain the _details_ for making this change. What existing problem does the pull request solve?**

This PR adds a week view and since the week view can be configured to show any number of days it also satisifys the day view requirement. In fixing this a broke the toolbar of the calendar to its own component and fixed related issues on that. This effects calendar, datepicker, monthview, weekview to test.

When testing this issue in QA it is better to raise new issues than fail the entire ticket for this change. But please note anything on the PR before its merged and we can decide

**Known to be Not Working / Descoped**

1) RTL date rendering http://localhost:4000/components/calendar/example-index.html?locale=ar-SA
2) IE 11 will not have sticky header
3) General Mobile view
4) When switching theme reload the page 

**Related github/jira issue (required)**:
Fixed #3080
Fixed #2862
Fixed #2780
Fixed #1757
Fixed #3068
Fixed #2962

**Steps necessary to review your pull request (required)**:
**3080**
- go to http://localhost:4000/components/monthview/example-index.html
- click the date picker in the header
- hit cancel
- the date should remain November 2019

**2862**
- open http://localhost:4000/components/monthview/example-index.html in IE 11
- make sure the header looks nice

**2780**
- go to http://localhost:4000/components/week-view/example-one-day.html
- go to http://localhost:4000/components/week-view/example-two-day.html
- go to http://localhost:4000/components/week-view/example-two-weeks.html
- they all show you can configure to show a day view
- can also get to it on http://localhost:4000/components/calendar/example-index.html
- on this page click any day with events and then select day view in the picker to see the day
- try things like themes and RTL ect 

**1757**
- go to http://localhost:4000/components/week-view 
- test all the examples here
- try things like themes and RTL ect 
- can also get to it on http://localhost:4000/components/calendar/example-index.html
- on this page click any day with events and then select day view in the picker to see the day

**3068*
- go to http://localhost:4000/components/datepicker/example-index.html?locale=ar-SA
- open the calendar
- the page should align nicely
- try IE
- also note that the apply button (azure) has changed to the other side

**2962**
- go to http://localhost:4000/components/calendar/example-index.html
- make the screen shorter
- click an event
- the modal should scroll now
- the arrow should now show up pointing to the event
- NOTE: the dialog stays open and does not resize when resizing pages. This is not supported

**Included in this Pull Request**:
